stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Any, Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self, TypeAlias
+
+_T = TypeVar("_T")
+_SendType: TypeAlias = Literal["send", "recv", "both"]
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+@final
+class ChannelID:
+    @property
+    def end(self) -> _SendType: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+
+@final
+class ChannelInfo(structseq[Any], tuple[bool, bool, bool, int, int, int, int, int]):
```

Comment:
Since `bool` is a subclass of `int`, and all elements in the tuple are either `bool` or `int`, this could maybe be

```suggestion
class ChannelInfo(structseq[int], tuple[bool, bool, bool, int, int, int, int, int]):
```